File: src/foundation/java/io/karma/pda/foundation/component/PlayerImage.java
/*
 * Copyright (C) 2024 Karma Krafts & associates
 */

package io.karma.pda.foundation.component;

import io.karma.pda.api.app.component.AbstractComponent;
import io.karma.pda.api.app.component.ComponentType;
import io.karma.pda.api.state.MutableState;
import io.karma.pda.api.state.Synchronize;
import io.karma.peregrine.api.color.Color;
import io.karma.peregrine.api.color.ColorProvider;

import java.util.UUID;

/**
 * @author Alexander Hinze
 * @since 13/04/2024
 */
public class PlayerImage extends AbstractComponent {
    @Synchronize
    public final MutableState<UUID> player = MutableState.of(UUID.fromString("069a79f4-44e9-4726-a5be-fca90e38aaf5"));
    @Synchronize
    public final MutableState<RenderType> renderType = MutableState.of(RenderType.HEAD_2D);
    @Synchronize
    public final MutableState<ColorProvider> background = MutableState.of(Color.NONE);
    @Synchronize
    public final MutableState<ColorProvider> foreground = MutableState.of(Color.NONE);

    public PlayerImage(final ComponentType<?> type, final UUID id) {
        super(type, id);
    }

    public enum RenderType {
        HEAD_2D, HEAD_3D, BODY_2D, BODY_3D
    }
}

File: src/composable/kotlin/io/karma/pda/composition/component/PlayerImage.kt
/*
 * Copyright (C) 2024 Karma Krafts & associates
 */

@file:JvmName("ComponentExtensionsKt")
@file:JvmMultifileClass

package io.karma.pda.composition.component

import io.karma.pda.api.app.component.Container
import io.karma.pda.composition.Composable
import io.karma.pda.composition.Composer
import io.karma.pda.composition.LayoutProps
import io.karma.pda.foundation.component.DefaultComponents
import io.karma.pda.foundation.component.PlayerImage

/**
 * @author Alexander Hinze
 * @since 07/06/2024
 */

@Composable
inline fun <C : Container> Composer<C>.PlayerImage(
    crossinline layoutProps: LayoutProps,
    localName: String? = null,
    props: @Composable PlayerImage.() -> Unit = {}
) = component(
    type = DefaultComponents.PLAYER_IMAGE,
    layoutProps = layoutProps,
    localName = localName,
    props = props
)